### ISSUE TITLE
Do enumerated devices really need to reference the enumerator?

### ISSUE THREAD

**Issue by main--** [dcuddeback/libudev-rs#16]
The compiler's lifetime defaulting appears to produce this:

```rust
impl<'a> Enumerator<'a> {
    fn scan_devices<'b>(&'b mut self) -> Result<Devices<'b>>;
}
```

Is that intentional? Naively, I would expect the enumerated devices to be valid even after the enumerator is gone. I didn't find any official libudev documentation about this however.

It sure would be a lot more convenient if this function returned a `Devices<'a>`.

**Comment by dcuddeback** [comment]
@main-- I believe the reason it references the enumerator is because the device list is a part of the enumerator data structure in the C library that I'm wrapping. It's been a while since I wrote that code, so I may be wrong.

Regardless, I agree that the lifetimes need to be revisited.

**Comment by ruabmbua** [comment]
Similar problem with the Device::parent() impl. The compiler infers a too conservative lifetime.

See https://users.rust-lang.org/t/lifetime-problem-with-libudev/28867/7.

**Comment by dcuddeback** [comment]
I think this should be solved by #13, which should be landing soon.

**Comment by dcuddeback** [comment]
This should be fixed in #13, which is now released in v0.3.0: https://github.com/dcuddeback/libudev-rs/releases/tag/v0.3.0.
